src/lib.rs

```diff
@@ -416,28 +491,41 @@ pub mod pallet {
 
         fn log_block_rewards(amount: BalanceOf<T>) {
             let mor_config = <MorConfigStorage<T>>::get();
-            let n_blocks = mor_config.time_period_blocks;
-            if !<RewardsRecord<T>>::exists() {
-                <RewardsRecord<T>>::set((1u8, vec![<BalanceOf<T>>::zero(); n_blocks as usize]));
-            }
+            let n_blocks = mor_config.track_n_block_rewards;
 
-            // RewardsRecord: (u8, Vec<BalanceOf<T>>)
-            // Next array-slot to write in, Array of imbalances
-            let (mut slot_cnt, mut balances) = <RewardsRecord<T>>::get();
+            // RewardsRecordStorage: (u8, Vec<BalanceOf<T>>)
+            // (Next array-slot to write in, vector of imbalances)
+            let (mut slot_cnt, mut balances) = <RewardsRecordStorage<T>>::get();
             balances[slot_cnt as usize] = amount;
             slot_cnt += 1;
             if slot_cnt >= n_blocks {
                 slot_cnt = 0;
             }
 
-            // PeriodReward: BalanceOf<T>
+            // PeriodRewardStorage: BalanceOf<T>
             // Sum of last n_blocks block-rewards
-            let mut period_reward = <BalanceOf<T>>::zero();
-            // Workarround, skip some block rewards to gain always positive balance
-            balances.iter().skip(5).for_each(|&b| period_reward += b);
+            let mut period_reward = BalanceOf::<T>::zero();
+            balances.iter().for_each(|&b| period_reward += b);
```

Comment:
I saw the previous implementation skipped five blocks, but now we don't have this limit. Could I ask why? Does that because we put the RewardsRecordStorage into genesis settings?

Reply:
Yes, it was kind of mixed effects and now we don't need this workaround anymore...